ast/dml.go

```diff
@@ -85,7 +85,51 @@ type Join struct {
 
 // Restore implements Node interface.
 func (n *Join) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	if err := n.Left.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while restore Join.Left")
+	}
+	ctx.WritePlain(" ")
```

Comment:
You could skip this extra call if you attach the space to the `JOIN`/`STRAIGHT_JOIN` keyword.

```go
if condition {
    ctx.WriteKeyWord(" NATURAL")
}
if condition {
    ctx.WriteKeyWord(" LEFT")
}
ctx.WriteKeyWord(" JOIN ")
```